src/utils.ts

```diff
@@ -7,10 +7,24 @@ import path from 'path';
 import stealthyRequire from 'stealthy-require';
 import { getNextConfig } from './nextConfig';
 import { InternalError } from './_error';
+import { normalizeLocalePath } from 'next/dist/next-server/lib/i18n/normalize-locale-path';
 
 export function parseRoute({ route }: { route: string }) {
```

Comment:
- `parseRoute` has the same signature as before and it's used in multiple places.
- `parseLocaleRoute` is now used from the `makeRouterMock` and in addition returns `detectedLocale` that is needed. I've split it into 2 functions to minimize amount of changes and because I still find the old signature useful -- when you don't care about locales and are only interested in the `urlObject`

Reply:
Got it. Let's keep it like this, we'll refactor `parseRoute` at the next opportunity. 👍

Reply:
(I've taken the liberty of adding a reminder)